src/tadox.ts

```diff
@@ -0,0 +1,115 @@
+import type {
+  ACMode,
+  FanLevel,
+  FanSpeed,
+  HorizontalSwing,
+  Power,
+  Termination,
+  VerticalSwing,
+  XRoomsAndDevices,
+} from "./types";
+
+import { Method } from "axios";
+import { Tado } from "./tado";
+import * as utils from "./utils";
+
+const tado_x_url = "https://hops.tado.com";
+
+export class TadoX extends Tado {
+  /**
+   * Makes an API call to the provided TadoX URL with the specified method and data.
+   *
+   * @typeParam R - The type of the response
+   * @typeParam T - The type of the request body
+   * @param url - The endpoint to which the request is sent. If the URL contains "https", it will be used as is.
+   * @param method - The HTTP method to use for the request (e.g., "get", "post").
+   * @param data - The payload to send with the request, if applicable.
+   * @returns A promise that resolves to the response data.
+   */
+  async apiCallX<R, T = unknown>(url: string, method: Method = "get", data?: T): Promise<R> {
+    const callUrl = tado_x_url + url;
```

Comment:
I'd recommend the base url be made a class property and then overridden in this class instead of overriding the `apiCall` as right not eiq calls will fail.

Reply:
I don't think the EnergyIQ calls fail because they still use the `apiCall` method as opposed to the `apiCallX` method.

Switching to class members is probably fair, they're not exported so having them top level has no advantages.

I don't think I can override the `tado_url` in the `TadoX` class though because then it'll be used for all call instead of just the new ones... unless I'm misunderstanding your suggestion.

Reply:
I restructured the tests to include both the `Tado` and `TadoX` classes for all high-level functions